src/instructlab/model/chat.py

```diff
@@ -789,18 +789,35 @@ def chat_cli(
     sys_prompt = CONTEXTS.get(context, "default")(get_model_arch(pathlib.Path(model)))
     loaded["messages"] = [{"role": "system", "content": sys_prompt}]
 
+    retriever: DocumentStoreRetriever | None = None
+
     # Instantiate retriever if RAG is enabled
     if rag_enabled:
         logger.debug("RAG enabled for chat; initializing retriever")
-        retriever: DocumentStoreRetriever | None = create_document_retriever(
-            document_store_uri=document_store_uri,
-            document_store_collection_name=collection_name,
-            top_k=top_k,
-            embedding_model_path=embedding_model_path,
-        )
+        try:
+            retriever: DocumentStoreRetriever | None = create_document_retriever(
```

Comment:
You already type-hinted retriever on line 792 so there's no reason to do it here, especially if you're assigning it a return value from another function.